src/test/regress/expected/view_propagation.out

```diff
@@ -855,6 +855,85 @@ HINT:  Use DROP MATERIALIZED VIEW to remove a materialized view.
+-- views will be created locally with warnings
+CREATE VIEW vv1 as SELECT * FROM local_t;
+WARNING:  "view vv1" has dependency to "table local_t" that is not in Citus' metadata
+DETAIL:  "view vv1" will be created only locally
+HINT:  Distribute "table local_t" first to distribute "view vv1"
+CREATE OR REPLACE VIEW vv2 as SELECT * FROM vv1;
+WARNING:  "view vv2" has dependency to "table local_t" that is not in Citus' metadata
+DETAIL:  "view vv2" will be created only locally
+HINT:  Distribute "table local_t" first to distribute "view vv2"
+-- view will fail due to local circular dependency
+CREATE OR REPLACE VIEW vv1 as SELECT * FROM vv2;
+ERROR:  Citus can not handle circular dependencies between distributed objects
+DETAIL:  "view vv1" circularly depends itself, resolve circular dependency first
+-- local view with no citus relation dependency will be distributed
+CREATE VIEW v_dist AS SELECT 1;
+-- show that the view is in pg_dist_object
+SELECT COUNT(*) FROM pg_dist_object WHERE objid = 'v_dist'::regclass;
+ count
+---------------------------------------------------------------------
+    1
+(1 row)
+
+-- tests when citus.enforce_object_restrictions_for_local_objects=false
+SET citus.enforce_object_restrictions_for_local_objects TO false;
+SET citus.enable_unsupported_feature_messages TO false;
+-- views will be created locally without errors OR warnings
+CREATE VIEW vv3 as SELECT * FROM local_t;
+CREATE OR REPLACE VIEW vv4 as SELECT * FROM vv3;
+CREATE OR REPLACE VIEW vv3 as SELECT * FROM vv4;
+-- show that views are NOT in pg_dist_object
+SELECT COUNT(*) FROM pg_dist_object WHERE objid IN ('vv3'::regclass, 'vv4'::regclass);
+ count
+---------------------------------------------------------------------
+    0
+(1 row)
+
+-- local view with no citus relation dependency will NOT be distributed
+CREATE VIEW v_local_only AS SELECT 1;
+-- show that the view is NOT in pg_dist_object
+SELECT COUNT(*) FROM pg_dist_object WHERE objid = 'v_local_only'::regclass;
+ count
+---------------------------------------------------------------------
+    0
+(1 row)
+
+-- distribute the local table and check the distribution of dependent views
+SELECT create_distributed_table('local_t', 'a');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+-- show that views with no circular dependency are in pg_dist_object
+SELECT COUNT(*) FROM pg_dist_object WHERE objid IN ('vv1'::regclass, 'vv2'::regclass);
+ count
+---------------------------------------------------------------------
+    2
+(1 row)
+
+-- show that views with circular dependency are NOT in pg_dist_object
+SELECT COUNT(*) FROM pg_dist_object WHERE objid IN ('vv3'::regclass, 'vv4'::regclass);
+ count
+---------------------------------------------------------------------
+    0
+(1 row)
+
+-- show that we cannot re-create the circular views ever
+CREATE OR REPLACE VIEW vv3 as SELECT * FROM local_t;
```

Comment:
maybe just for future cases, can you re-run `CREATE OR REPLACE VIEW vv3 as SELECT * FROM vv4;` after  `RESET citus.enforce_object_restrictions_for_local_objects;` as well?

So that we never introduce weird bugs